src/components/structures/RoomView.js

```diff
@@ -1449,6 +1450,9 @@ module.exports = React.createClass({
         // when waiting for a room to be returned by js-sdk when joining
         const previewBarSpinner = this.state.joining || this.state.waitingForRoom;
 
+        const config = SdkConfig.get();
+        const hideScalar = !config.integrations_ui_url || !config.integrations_rest_url;
```

Comment:
`RoomView` doesn't strictly need to "know" that scalar is disabled. Maybe we can shift this to a static function on `ScalarClient` and call it `isScalarConfigured`.